devices/src/virtio/queue.rs

```diff
@@ -327,7 +327,7 @@ impl Queue {
 
         let used_ring = self.used_ring;
         let next_used = (self.next_used.0 % self.actual_size()) as usize;
-        let used_elem = used_ring.unchecked_add(4 + next_used * 8);
+        let used_elem = used_ring.unchecked_add((4 + next_used * 8) as u64);
```

Comment:
It looks like the `as` is no longer necessary if we remove the `as usize` from the initialization of `next_used`.